MarsRover/Grid.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace MarsRover
+{
+    public class Grid
+    {
+        private int _xCoordinates;
+        private int _yCoordinates;
+        private int _gridSizeX;
+        private int _gridSizeY;
+
+
+        public Grid(int x, int y, int sizeXGrid, int sizeYGrid)
+        {
+            _xCoordinates = Math.Abs(x);
+            _yCoordinates = Math.Abs(y);
+            _gridSizeX = Math.Abs(sizeXGrid);
+            _gridSizeY = Math.Abs(sizeYGrid);
+        }
+
+        public Grid GridEntryAndSizeValidation(Grid grid)
+        {
+            if (grid._xCoordinates > grid._gridSizeX || grid._yCoordinates > grid._gridSizeY)
+            {
+                grid._xCoordinates = 0;
+                grid._yCoordinates = 0;
+            }
+
+            return grid;
+        }
+
+        public int XCoordinate() => _xCoordinates;
```

Comment:
So studying the usage of this and the y-coordinate in this class, it appears to be tracking the location of the rover on the grid.  I would suggest that the grid shouldn't know about where on the grid the rover is, but that the rover itself should store its own location.  Think of the refactoring involved if we needed to add a second or third rover to this kata!

Reply:
Maybe we can talk about this one so I get a better understanding.  Each `Rover` gets a reference to it's own `Grid` to track movement so I believe the `RoverClient` could technically create as many rovers as it wanted and each rover would still have unique x and y coordinates.